apps/src/lab2/lab2Redux.ts

```diff
@@ -40,6 +40,13 @@ export interface LabState {
   isLoadingProjectOrLevel: boolean;
   // If the lab is loading. Can be updated by lab-specific components.
   isLoading: boolean;
+  // Error currently on the page, if present. If not undefined, isPageError should also be true.
```

Comment:
It always feels a bit risky when two state variables need to stay in sync; if anything goes wrong here, we could have four potential states.  Is it possible to unify them?

Reply:
Yeah, good call. I'll update the logic to check if pageError is undefined, as per Molly's suggestion